library/network_connections.py

```diff
@@ -993,7 +995,8 @@ def connection_create(self, connections, idx, connection_current=None):
                     s_ip4.add_dns(d["address"])
             for s in ip["dns_search"]:
                 s_ip4.add_dns_search(s)
-
+            for s in ip["dns_options"]:
+                s_ip4.add_dns_option(s)
```

Comment:
dns-options property is a list of strings.

Most properties that are a list of something don't differentiate between an empty list and an unset list. However, for dns-options that makes a difference:

* leaving the property unset (the default), means that the DNS options can be configured as default values in NetworkManager.conf
* leaving the property explicitly set to an empty list, means that there are not DNS options set (and the defaults are not used).

The role should possibly fully define a profile. At least, for the properties that it supports. Hence, you should first once call `s_ip4.clear_dns_options(True)` so that the profile has NO DNS options set.


For initscripts, that difference doesn't exist. However, we often want that an ifcfg file also works when using it in NetworkManager. For that reason, it might make sense to explicitly set `ifcfg["RES_OPTIONS"] = ''`. But that is less clearly the desired things...

Reply:
updated, thanks